tests/integration/test_encrypted_disk/test.py

```diff
@@ -580,3 +584,127 @@ def test_backup_restore(
     else:
         node.query(restore_command)
         assert node.query(select_query) == "(0,'data'),(1,'data')"
+
+
+def _create_aws_kms_key(key):
+    # Create KMS encryption key
+    headers = {
+        "Content-Type": "application/json",
+        "X-Amz-Target": "TrentService.CreateKey",
+    }
+    res = requests.post(cluster.local_kms_url, headers=headers)
+    assert res.status_code == 200
+    res_json = json.loads(res.text)
+    key_arn = res_json["KeyMetadata"]["Arn"]
+    key_id = res_json["KeyMetadata"]["KeyId"]
+
+    # Encrypt secret key with KMS
+    headers["X-Amz-Target"] = "TrentService.Encrypt"
+    res = requests.post(
+        cluster.local_kms_url,
+        headers=headers,
+        json={"KeyId": key_id, "Plaintext": base64.b64encode(key.encode()).decode()},
+    )
+    assert res.status_code == 200
+    res_json = json.loads(res.text)
+    key_encrypted_base64 = res_json["CiphertextBlob"]
+
+    # Decrypt secret key wih KMS to ensure Local KMS works as expected
+    headers["X-Amz-Target"] = "TrentService.Decrypt"
+    res = requests.post(
+        cluster.local_kms_url,
+        headers=headers,
+        json={"KeyId": key_id, "CiphertextBlob": key_encrypted_base64},
+    )
+    assert res.status_code == 200
+    res_json = json.loads(res.text)
+    key_decrypted = res_json["Plaintext"]
+    assert key == base64.b64decode(key_decrypted).decode()
+
+    return key_arn, key_encrypted_base64
+
+
+# Test adding AWS encryption key on the fly.
+def test_add_aws_keys():
+    key1 = "firstfirstfirstf"
+    keys = f"<key>{key1}</key>"
+    make_storage_policy_with_keys(
+        "encrypted_policy_multikeys", keys, check_system_storage_policies=True
+    )
+
+    # Add some data to an encrypted disk.
+    node.query(
+        """
+        CREATE TABLE encrypted_test (
+            id Int64,
+            data String
+        ) ENGINE=MergeTree()
+        ORDER BY id
+        SETTINGS storage_policy='encrypted_policy_multikeys'
```

Comment:
does this policy use fs cache? if so, we won't read from s3 all the time.

Reply:
It doesn't go to S3 at all. This test doesn't need to. Check out the [make_storage_policy_with_keys](https://github.com/ClickHouse/ClickHouse/blob/7721d4fb37e215be2ddbd2366e64312708387061/tests/integration/test_encrypted_disk/test.py#L175-L221). It generates a policy that uses `disk_local`